Frameworks/Foundation/NSDecimal.mm

```diff
@@ -0,0 +1,592 @@
+    }
+
+    // same length
+    int index = std::min(LENGTH(leftOperand), LENGTH(rightOperand)) - 1;
+    while (index >= 0) {
+        if (MANTISSA(leftOperand)[index] < MANTISSA(rightOperand)[index]) {
+            return (IS_NEGATIVE(rightOperand)) ? NSOrderedDescending : NSOrderedAscending;
+        }
+
+        if (MANTISSA(leftOperand)[index] > MANTISSA(rightOperand)[index]) {
+            return IS_NEGATIVE(rightOperand) ? NSOrderedAscending : NSOrderedDescending;
+        }
+        index--;
+    }
+
+    if (LENGTH(rightOperand) > LENGTH(leftOperand)) {
+        return IS_NEGATIVE(rightOperand) ? NSOrderedDescending : NSOrderedAscending;
+    }
+    if (LENGTH(rightOperand) < LENGTH(leftOperand)) {
+        return IS_NEGATIVE(rightOperand) ? NSOrderedAscending : NSOrderedDescending;
+    }
+
+    return NSOrderedSame;
+}
+
+/**
+ @Status Interoperable
+*/
+void NSDecimalCopy(NSDecimal* destination, const NSDecimal* source) {
+    if (!source || !destination) {
+        return;
+    }
+
+    NEGATIVE_FLAG(destination) = NEGATIVE_FLAG(source);
+    EXPONENT(destination) = EXPONENT(source);
+    LENGTH(destination) = LENGTH(source);
+    destination->_isCompact = source->_isCompact;
+    destination->_reserved = source->_reserved;
+    memcpy(&MANTISSA(destination), &MANTISSA(source), NSDecimalMaxSize * sizeof(unsigned short));
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalDivide(NSDecimal* result,
+                                   const NSDecimal* leftOperand,
+                                   const NSDecimal* rightOperand,
+                                   NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Stub
+ @Notes
+*/
+NSCalculationError NSDecimalMultiply(NSDecimal* result,
+                                     const NSDecimal* leftOperand,
+                                     const NSDecimal* rightOperand,
+                                     NSRoundingMode roundingMode) {
+    UNIMPLEMENTED();
+    return StubReturn();
+}
+
+/**
+ @Status Interoperable
+*/
+NSCalculationError NSDecimalMultiplyByPowerOf10(NSDecimal* result, const NSDecimal* number, short power, NSRoundingMode roundingMode) {
+    THROW_NS_IF_FALSE(E_INVALIDARG, (result != NULL) || (number != NULL));
+
+    NSDecimalCopy(result, number);
+    int exponent;
+    if ((exponent = EXPONENT(result) + power) > EXPONENT_MAX) {
+        INVALIDATE_DECIMAL(result);
+        return NSCalculationOverflow;
+    }
+
+    if (exponent < EXPONENT_MIN) {
+        INVALIDATE_DECIMAL(result);
+        return NSCalculationUnderflow;
+    }
+
+    EXPONENT(result) = exponent;
+    return NSCalculationNoError;
+}
+
+/**
+ @Status Interoperable
+*/
+NSCalculationError NSDecimalNormalize(NSDecimal* number1, NSDecimal* number2, NSRoundingMode roundingMode) {
+    THROW_NS_IF_FALSE(E_INVALIDARG, (number1 != NULL) || (number2 != NULL));
```

Comment:
We should have tests confirming this behaviour

---
In reply to: [75561368](https://github.com/Microsoft/WinObjC/pull/896#discussion_r75561368) [](ancestors = 75561368,75560872)